api/userpassloginprovider.go

```diff
@@ -57,6 +57,12 @@ type userpassLoginProvider struct {
 	cookieURL    *url.URL
 }
 
+// Token implements the [LoginProvider.Token] method.
+// Returns the user's password.
+func (p *userpassLoginProvider) Token() (string, error) {
+	return p.password, nil
```

Comment:
Also here - there's no check that password can't be set to "" and the contract of the Token() method would be violated

Reply:
Right, with the change I mentioned above to avoid checking for `""` , I think it would be fine to allow empty strings here.